master/contrib/github_buildbot.py

```diff
@@ -143,7 +144,27 @@ def render_POST(self, request):
             request.setResponseCode(INTERNAL_SERVER_ERROR)
             return json.dumps({"error": e.message})
 
-    def process_change(self, payload, user, repo, repo_url, project, request):
+    def process_change(self, change, branch, repo, repo_url):
+        files = change['added'] + change['removed'] + change['modified']
+        who = ""
+        if 'username' in change['author']:
+            who = change['author']['username']
+        else:
+            who = change['author']['name']
+        if 'email' in change['author']:
+            who = "%s <%s>" % (who, change['author']['email'])
+
+        return \
+            {'revision': change['id'],
+             'revlink': change['url'],
+             'who': who,
+             'comments': change['message'][:1024],
```

Comment:
I think this might be a more general bug, as the sqlite database doesn't enforce column length limits. Changed to include a message.